docs/docs/porting-a-jekyll-website-to-gatsby.md

```diff
@@ -0,0 +1,276 @@
+  └── index.md
+```
+
+### Migrating the markdown / Liquid templates pages
+
+Create a new folder in Gatsby project under the `src/pages-markdown` for those pages written in markdown format. I left the `src/pages` as default so in future we need it for some javaScript files. Now its time to transfer all markdown files to `src/pages-markdown` like this.
+
+```
+  src
+  └── pages-markdown
+      ├── footer.md
+      └── header.md
+```
+
+In addition, these pages must be have next properties within the `frontmatter` section:
+
+```md
+---
+layout: page
+title: About
+date: 2010-11-23 14:48
+path: /about
+---
+```
+
+Now its time to make Gatsby engine take account into these files to inform where to read them:
+
+```json
+// gatsby-config.js
+{
+  "resolve": `gatsby-source-filesystem`,
+  "options": {
+    "name": `markdown-pages`,
+    "path": `${__dirname}/src/pages-markdown/`
+  }
+}
+```
+
+For each file found at `src/pages-markdown` Gatsby will create a `node`. We can extend the `createPages` method and make Gatsby create a new page for each `node`. The steps can be summarized as:
+In addition, these pages must be have next properties within the `frontmatter` section:
+
+- Query for all the markdown nodes read by gatsby
+- Filter those nodes with `layout=page`
+- Create a new page invoking `action.createPage` method and using the template file src/templated/page.js
+
+```js
+// gatsby-node.js
+exports.createPages = ({ graphql, actions }) => {
+  const { createPage } = actions
+  const pageTemplate = path.resolve(`src/templates/page.js`)
+
+  return graphql(
+    `
+      {
+        allMarkdownRemark {
+          edges {
+            node {
+              fields {
+                slug
+              }
+              frontmatter {
+                path
+                layout
+              }
+            }
+          }
+        }
+      }
+    `
+  ).then(result => {
+    if (result.errors) {
+      return Promise.reject(result.errors)
+    }
+
+    const markdownItems = result.data.allMarkdownRemark.edges
+
+    // Create pages and blog post pages
+    markdownItems.forEach(({ node }) => {
+      if (node.frontmatter.layout === "page") {
+        createPage({
+          path: node.frontmatter.path,
+          component: pageTemplate,
+        })
+      }
+    })
+  })
+}
+```
+
+### Migrating blog posts
+
+Now its time to transfer all the blog posts in to `src/content` folder.
+
+```
+  src
+  └── content
+      ├── 2007-10-29-why-every-programmer-should-play-nethack.md
+      └── 2009-04-26-barcamp-boston-4-roundup.md
+```
+
+We need to tell gatsby where to find blog posts.
+
+```js
+// gatsby-config.js
+{
+  resolve: 'gatsby-source-filesystem',
+  options: {
+    name: 'blog',
+    path: `${__dirname}/blog/`,
+  },
+},
+```
+
+In Jekyll when we create a new blog post we create something like that `YYYY-MM-DD-some-title.md` so we need to create that `slug` from the file name. The way we can do is using the `onCreateNode` method
+
+```js
```

Comment:
```suggestion
```js:title=gatsby-node.js
```

Reply:
Should we recommend people use regex? Or rename their files? Might be worth mentioning both options.